src/pe/debug.rs

```diff
@@ -140,12 +355,6 @@ impl<'a> CodeviewPDB70DebugInfo<'a> {
         idd: &ImageDebugDirectory,
         opts: &options::ParseOptions,
     ) -> error::Result<Option<Self>> {
-        if idd.data_type != IMAGE_DEBUG_TYPE_CODEVIEW {
-            // not a codeview debug directory
-            // that's not an error, but it's not a CodeviewPDB70DebugInfo either
-            return Ok(None);
-        }
-
```

Comment:
i'm just curious why did this get removed? Is it because we now parse CODEVIEW?

Reply:
The check is moved to the caller:

```rs
if let Some(idd) = &it.find_type(IMAGE_DEBUG_TYPE_CODEVIEW) {
    codeview_pdb70_debug_info = CodeviewPDB70DebugInfo::parse_with_opts(bytes, idd, opts)?;
    codeview_pdb20_debug_info = CodeviewPDB20DebugInfo::parse_with_opts(bytes, idd, opts)?;
}
```

Consumers (callers) should be aware of debug types whenever they are calling any of `IMAGE_DEBUG_TYPE_*` parsers.